### PR TITLE
Add utility to compare poses between two sparse models

### PR DESCRIPTION
**Pull request by anmatako**
`model_comparator` takes two sparse models as input and after aligning them it calculates some simple statistics on difference in pose parameters (translation, rotation, projection center) between them.

### CODE REVIEW

File: src/exe/colmap.cc [hunk 1]
```diff
@@ -1256,6 +1256,98 @@ int RunModelAnalyzer(int argc, char** argv) {
...
+  std::sort(vals.begin(), vals.end());
+  std::cout << "Min:    " << vals[0] << std::endl;
+  std::cout << "Max:    " << vals[len - 1] << std::endl;
+  std::cout << "Mean:   " << avg << std::endl;
+  std::cout << "Median: " << vals[len / 2] << std::endl;
+  std::cout << "P90:    " << vals[size_t(0.9 * len)] << std::endl;
+  std::cout << "P99:    " << vals[size_t(0.99 * len)] << std::endl;
+}
+
+int RunModelComparator(int argc, char** argv) {
```
Comment (ahojnnes) [inline src/exe/colmap.cc:1270]: ModelComparer?

File: src/exe/colmap.cc [hunk 2]
```diff
@@ -1256,6 +1256,98 @@ int RunModelAnalyzer(int argc, char** argv) {
...
+  std::cout << StringPrintf("Points: %d", reconstruction2.NumPoints3D())
+            << std::endl;
+
+  PrintHeading1("Comparing reconstruction image poses");
+  auto common_ids = reconstruction1.FindCommonRegImageIds(reconstruction2);
+  std::cout << StringPrintf("Common images: %d", common_ids.size())
+            << std::endl;
+
+  Eigen::Matrix3x4d alignment;
+  const double kMinInlierObservations = 0.3;
```
Comment (ahojnnes) [inline src/exe/colmap.cc:1301]: It would be great to make these two parameters command-line options.

Comment (ahojnnes) [reply src/exe/colmap.cc:1301]: Using AddDefaultOption()

**Comment by ahojnnes** [comment]
It could be useful to not just print percentiles but provide optional output CLI parameters to specify paths to text files where we write all rotation, translation errors. It would allow for easy visualization outside of COLMAP in Python, etc. Thanks.

**Comment by anmatako** [comment]
> It could be useful to not just print percentiles but provide optional output CLI parameters to specify paths to text files where we write all rotation, translation errors. It would allow for easy visualization outside of COLMAP in Python, etc. Thanks.

Recommended changes were added in the new update

File: src/exe/colmap.cc [hunk 3]
```diff
@@ -1256,6 +1256,146 @@ int RunModelAnalyzer(int argc, char** argv) {
...
+                           const std::vector<double>& proj_center_errors) {
+  std::ofstream file(path, std::ios::trunc);
+  CHECK(file.is_open()) << path;
+
+  file.precision(17);
+  file << "# Model comparison pose errors; one entry per common image"
+       << std::endl;
+  file << "# <rotation error (deg)>, <translation error>, <proj center error>"
+       << std::endl;
+  for (int i = 0; i < rotation_errors.size(); ++i) {
```
Comment (ahojnnes) [inline src/exe/colmap.cc:1271]: size_t

Comment (ahojnnes) [reply src/exe/colmap.cc:1271]: Could benefit from `CHECK_EQ(rotation_errors.size(), translation_errors.size());` and `CHECK_EQ(rotation_errors.size(), proj_center_errors.size());`

**Comment by ahojnnes** [comment]
I made some cosmetic changes, which I thought were easier, if I directly do them instead of going through review comments. Closing this PR. Thanks!

**Pull request closed without merging**